backend/curriculum_tracking/tests/test_activity_log_entry_creators_envoked.py

```diff
@@ -106,3 +106,45 @@ def test_add_review(self):
             start_url, data={"status": COMPETENT, "comments": "blah more stuff"}
         )
         self.assertEqual(LogEntry.objects.count(), 2)
+
+
+class log_card_review_requested_Tests(APITestCase, APITestCaseMixin):
+    LIST_URL_NAME = "agilecard-list"  #based on class of api-viewset
+    SUPPRESS_TEST_POST_TO_CREATE = True
+    SUPPRESS_TEST_GET_LIST = True
+
+    def test_review_requested(self):
+        actor_user = UserFactory(is_superuser=True, is_staff=True)
+        content_item=factories.ProjectContentItemFactory(
+                project_submission_type=ContentItem.LINK, template_repo=None
+            )
+        card = factories.AgileCardFactory(
+            status=AgileCard.READY, 
+
```

Comment:
Ah, I see what happened, I pushed the changes to the testing file I made but not to the api_views file (which weren't so much changes as removing comments and breakpoints made for my own reference) 
Will push those changes.
Will check into using black too. 

In terms of this:
"This shows a misunderstanding. Logging in is only needed when interacting with the api. card.start_project doesn't touch apis."
Was from pattern matching on other tests - but I see now that I can set the card to being IP already when it's created, so will do that
Thanks for the feedback :)

Reply:
@sheenarbw 
I tried writing the test without card.start_project() (by creating a card that was already in progress)
but got an error according to an assert statement in the model file, i.e:

File "/Users/sam/Documents/Umuzi/Products/Tilde/backend/curriculum_tracking/models.py", line 709, in request_review
    assert (
AssertionError: cannot request a review if the project isn't started - P: something awesome part 173 - foo.681@example.com

So leaving as it was for now so that the tests don't fail, but keen to see how else it could be done?

Reply:
You can check the code that raises the exception. See what conditions it checks?